### PR TITLE
Add constraints to the combination of load and release

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #13118 
/kind enhancement
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

**Comment by xige-16** [comment]
The skiped test case will be reopened after issue #12952 fixed

**Comment by yanliang567** [comment]
let's update the case accordingly after the feature is ready

File: internal/querycoord/impl.go [hunk 1]
```diff
@@ -89,16 +89,18 @@ func (qc *QueryCoord) GetStatisticsChannel(ctx context.Context) (*milvuspb.Strin
...
 	if qc.stateCode.Load() != internalpb.StateCode_Healthy {
 		status.ErrorCode = commonpb.ErrorCode_UnexpectedError
-		err := errors.New("query coordinator is not healthy")
+		err := errors.New("QueryCoord is not healthy")
 		status.Reason = err.Error()
-		log.Debug("show collection end with query coordinator not healthy")
+		log.Warn("show collection failed", zap.String("role", typeutil.QueryCoordRole), zap.Int64("msgID", req.Base.MsgID), zap.Error(err))
 		return &querypb.ShowCollectionsResponse{
 			Status: status,
 		}, nil
```
Comment (jaime0815) [inline internal/querycoord/impl.go:106]: use err as a return value

File: internal/proxy/proxy_test.go [hunk 2]
```diff
@@ -1545,7 +1545,7 @@ func TestProxy(t *testing.T) {
 			Type:           milvuspb.ShowType_InMemory,
 		})
 		assert.NoError(t, err)
-		assert.Equal(t, commonpb.ErrorCode_Success, resp.Status.ErrorCode)
+		assert.Equal(t, commonpb.ErrorCode_UnexpectedError, resp.Status.ErrorCode)
```
Comment (jaime0815) [inline internal/proxy/proxy_test.go:1548]: Change UnexpectedError to ErrorCode_Unknown?

File: internal/querycoord/impl.go [hunk 3]
```diff
@@ -145,20 +160,48 @@ func (qc *QueryCoord) ShowCollections(ctx context.Context, req *querypb.ShowColl
...
 		ErrorCode: commonpb.ErrorCode_Success,
 	}
 	if qc.stateCode.Load() != internalpb.StateCode_Healthy {
 		status.ErrorCode = commonpb.ErrorCode_UnexpectedError
-		err := errors.New("query coordinator is not healthy")
+		err := errors.New("QueryCoord is not healthy")
 		status.Reason = err.Error()
-		log.Debug("load collection end with query coordinator not healthy")
+		log.Warn("load collection failed", zap.String("role", typeutil.QueryCoordRole), zap.Int64("msgID", req.Base.MsgID), zap.Error(err))
 		return status, nil
```
Comment (jaime0815) [inline internal/querycoord/impl.go:176]: using err as a return value

File: internal/querycoord/impl.go (hunk 1 above)
Comment (xige-16) [reply internal/querycoord/impl.go:106]: Modifying common proto requires too much code to change in  all components, and if it is really necessary, it is recommended to make a separate pr.

File: internal/proxy/proxy_test.go (hunk 2 above)
Comment (xige-16) [reply internal/proxy/proxy_test.go:1548]: Modifying common proto requires too much code to change in all components, and if it is really necessary, it is recommended to make a separate pr.

File: internal/querycoord/impl.go (hunk 1 above)
Comment (xige-16) [reply internal/querycoord/impl.go:106]: error is only used to receive grpc information

File: internal/querycoord/impl.go (hunk 3 above)
Comment (xige-16) [reply internal/querycoord/impl.go:176]: error is only used to receive grpc information

File: internal/querycoord/impl.go [hunk 4]
```diff
@@ -309,57 +408,87 @@ func (qc *QueryCoord) ShowPartitions(ctx context.Context, req *querypb.ShowParti
...
+			}
 
-		if len(partitionIDsToLoad) == 0 {
-			log.Debug("loadPartitionRequest completed", zap.String("role", typeutil.QueryCoordRole),
-				zap.Int64("msgID", req.Base.MsgID), zap.Int64("collectionID", req.CollectionID),
-				zap.Int64s("partitionIDs", partitionIDs))
-			return status, nil
+			// if part of partitions to load have been loaded by load partitions request, return error
+			// should release partitions first, then load partitions again
+			status.ErrorCode = commonpb.ErrorCode_UnexpectedError
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/impl.go]: fixed

File: internal/querycoord/impl.go [hunk 5]
```diff
@@ -400,86 +541,162 @@ func (qc *QueryCoord) ReleasePartitions(ctx context.Context, req *querypb.Releas
...
 	}
 
-	req.PartitionIDs = toReleasedPartitions
+	var releaseTask task
 	baseTask := newBaseTask(qc.loopCtx, querypb.TriggerCondition_GrpcRequest)
-	releasePartitionTask := &releasePartitionTask{
-		baseTask:                 baseTask,
-		ReleasePartitionsRequest: req,
-		cluster:                  qc.cluster,
+	if len(inMemoryPartitionsAfterRelease) == 0 {
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/impl.go]: fixed

**Pull request merged**